astropy/io/fits/column.py

```diff
@@ -112,6 +112,18 @@
                               r'(?:(?P<widthf>[0-9]+)\.'
                               r'(?P<precision>[0-9]+))?)')
 
+# TDISPn for both ASCII and Binary tables
+TDISP_RE_DICT = {'A':re.compile(r'(?:(?P<formatc>[AL])(?P<width>[0-9]+)+)|'),
+            'IBOZ':re.compile(r'(?:(?P<formatc>[IBOZ])(?:(?P<width>[0-9]+)'
+                      r'(?:\.{0,1}(?P<precision>[0-9]+))?))|'),
+            'F':re.compile(r'(?:(?P<formatc>[F])(?:(?P<width>[0-9]+)\.{1}'
+                      r'(?P<precision>[0-9])+)+)|'),
+            'EGD':re.compile(r'(?:(?P<formatc>[EGD])(?:(?P<width>[0-9]+)\.'
+                      r'(?P<precision>[0-9]+))+)'
+                      r'(?:E{0,1}(?P<exponential>[0-9]+)?)|'),
+            'ENES':re.compile(r'(?:(?P<formatc>E[NS])(?:(?P<width>[0-9]+)\.{1}'
+                      r'(?P<precision>[0-9])+)+)')}
+
```

Comment:
Rather than the complicated `if` structure below, I would just add the individual formats here, repeating them as necessary, i.e.,
```
TDISP_RE_DICT = {}
TDISP_RE_DICT['A'] = TDISP_RE_DICT['L'] = ...
etc.
```
(as a bonus, this should be faster and take less memory, since only references to the compiled regular expressions are stored)

Reply:
Will do.  What's the advantage to setting up the emtpy dict first, instead of just initially filling it with the singles? (i.e. F and A in the case)

Reply:
No advantage whatsoever! You can just fill with the known cases and add duplicates after.